ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.ts

```diff
@@ -58,31 +81,130 @@ export class ActivityFileAnswer implements OnDestroy {
+        config.data = {
+            title: 'SDK.FileUpload.ConfirmReuploadTitle',
+            confirmBtnText: 'SDK.FileUpload.ReuploadBtnText',
+            cancelBtnText: 'SDK.FileUpload.CancelBtnText'
+        };
+
+        const dialogRef = this.dialog.open(ConfirmDialogComponent, config);
+        dialogRef.afterClosed().subscribe((confirmUpload: boolean) => {
+            if (confirmUpload) {
+                this.uploadFile();
+            }
+        });
+    }
+
+    undoUploadedFile(): void {
+        const panelClass = 'file-upload-confirm-dialog';
+        const config = this.modalService.getDialogConfig(this.undoUploadButtonRef, panelClass);
+        config.data = {
+            title: 'SDK.FileUpload.ConfirmUndoUploadTitle',
+            confirmBtnText: 'SDK.FileUpload.UndoBtnText',
+            cancelBtnText: 'SDK.FileUpload.CancelBtnText'
+        };
+
+        const dialogRef = this.dialog.open(ConfirmDialogComponent, config);
+        dialogRef.afterClosed().subscribe((confirmUndo: boolean) => {
+            if (confirmUndo) {
+                this.patchAnswer(null);
+                this.setUploadedFile(null);
+            }
+        });
+    }
+
+    ngOnDestroy(): void {
+        this.ngUnsubscribe.next();
+        this.ngUnsubscribe.complete();
+    }
+
+    private initFileRequirements(): void {
+        this.fileMaxSize = this.block.maxFileSize;
+        this.fileMimeTypes = this.block.mimeTypes || [];
+    }
+
+    private initUploadedFile(): void {
+        if (this.block?.answer) {
+            this.setUploadedFile({
+                uploadGuid: '',
+                uploadUrl: ',',
+                name: this.block.answer.fileName,
+                size: this.block.answer.fileSize,
+                isUploaded: true
+            } as UploadFile);
+        }
+    }
 
-        this.fileUploadService.uploadFile(this.fileUploadData.uploadUrl, formData, this.file.type)
+    private uploadFile(): void {
+        this.fileUploadService.uploadFile(this.fileToUpload.uploadUrl, this.selectedFile)
             .pipe(
                 catchError(err => {
                     this.logger.logDebug('ActivityFileAnswer uploadFile error:', err);
                     return EMPTY;
                 }),
                 takeUntil(this.ngUnsubscribe)
             )
-            .subscribe((res) => {
-                this.block.answer = this.fileUploadData.uploadGuid;
-                this.valueChanged.emit(this.fileUploadData.uploadGuid);
+            .subscribe(() => {
+                this.patchAnswer(this.fileToUpload);
+                this.setUploadedFile({
+                    ...this.fileToUpload,
+                    isUploaded: true
+                });
+                this.resetSelectedFile();
+                this.errorMessage = '';
             });
     }
 
-    ngOnDestroy(): void {
-        this.ngUnsubscribe.next();
-        this.ngUnsubscribe.complete();
+    private setUploadedFile(file: UploadFile | null): void {
```

Comment:
#minor I'm not sure we need this set method... are we going to extend it with more functionality? if not probably it's fine to use the explicit set in the code

Reply:
let's keep it as is